common/src/enums/sponsorshipStatus.ts

```diff
@@ -0,0 +1,8 @@
+export enum SponsorshipStatus {
+  None = 0,
+  Pending = 1,
+  Sent = 2,
```

Comment:
I would prefer these to be just something like `offered`

I'm not sure when `none` would be used since the row just wouldn't exist, `pending` would be a transient state after create and before sending the offer, and `Sent` would require updating the model after emailing, but I'd prefer not to have that be part of a transaction with the DB. It's why we have resend commands

I think you might be trying to expand these states a little more than I imagined to accommodate self-hosted states where things have "happened" but haven't been synced. I think it's a good enough experience to just have the state lag behind by 1 sync, right?